### PR TITLE
Create ReadOnlySpan marshaller for ManagedToUnmanagedOut scenarios

### LINKED ISSUE #96525: Returning a ReadOnlySpan<byte> from a LibraryImport fails
### Description

Returning a ReadOnlySpan<byte> from a call generated by LibraryImport fails to compile, even though it should. It correctly works with a byte[] as the return type instead of ReadOnlySpan<byte>

### Reproduction Steps

Compile the following code definition

```
    [LibraryImport("wpiutil", EntryPoint = "WPI_GetResource_wpilib_128_png")]
    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
    [return: MarshalUsing(CountElementName = nameof(len))]
    public static unsafe partial ReadOnlySpan<byte> GetResourceWpilib128Png(out nuint len);
```

### Expected behavior

The code should compile

### Actual behavior

The code fails with the following error 

```
C:\Users\thadh\Documents\GitHub\robotdotnet\WPILib\src\wpiutil\Natives\ImagesNative.cs(13,53): error SYSLIB1051: The specified parameter needs to be marshalled from unmanaged to managed, but the marshaller type 'global::System.Runtime.InteropServices.Marshalling.ReadOnlySpanMarshaller<,>' does not support it. The generated source will not handle marshalling of the return value of method 'GetResourc
eWpilib128Png'. (https://learn.microsoft.com/dotnet/fundamentals/syslib-diagnostics/syslib1051) [C:\Users\thadh\Documents\GitHub\robotdotnet\WPILib\src\wpiutil\wpiutil.csproj]
C:\Users\thadh\Documents\GitHub\robotdotnet\WPILib\src\wpiutil\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs(574,68): error CA1420: Managed parameter or return types require runtime marshalling to be enabled (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1420) [C:\Users\thadh\Documents\GitHub\robotdotnet\WPILib\ 
src\wpiutil\wpiutil.csproj]
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

.NET 8

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jtschuster**
Adds a new marshaller for `ReadOnlySpan` to marshal in `ManagedToUnmanagedOut` scenarios to enable returning a `ReadOnlySpan` from a `LibraryImport` method.

Fixes https://github.com/dotnet/runtime/issues/96525

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
This will need an API proposal as it does introduce new public API. Can you make an issue for that/update the issue that this is fixing?

**Comment by jtschuster** [comment]
API proposal: https://github.com/dotnet/runtime/issues/101136

**Comment by jtschuster** [comment]
@dotnet/interop-contrib API got approved, please take a look when you have a chance.

**Review by jkoritzinsky — DISMISSED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ReadOnlySpanMarshaller.cs [hunk 1]
```diff
@@ -166,5 +167,58 @@ public void Free()
...
+                _managedValues = new T[numElements];
+                return _managedValues;
+            }
+
+            /// <summary>
+            /// Frees resources.
+            /// </summary>
+            public void Free()
+            {
+                NativeMemory.Free(_unmanagedArray);
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ReadOnlySpanMarshaller.cs:220]: I think that this should be `Marshal.FreeCoTaskMem((IntPtr)_unmanagedArray);`

@jkoritzinsky Can you please confirm.

Comment (jkoritzinsky) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ReadOnlySpanMarshaller.cs:220]: Yep this should be `FreeCoTaskMem`. Good catch!

**Review by jkoritzinsky — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**